lib/vega_lite/data.ex

```diff
@@ -171,12 +169,20 @@ defmodule VegaLite.Data do
     build_fields(fields, root, cols)
   end
 
+  defp used_fields(fields, nil), do: used_fields(fields)
+  defp used_fields(fields, extra_fields), do: used_fields(fields) ++ [extra_fields]
+
   defp used_fields(fields) do
     for {_key, field} <- fields, uniq: true do
       if is_list(field), do: field[:field], else: field
     end
   end
 
+  defp build_options(data, fields) do
+    {extra_fields, fields} = Keyword.pop(fields, :extra_fields)
+    {columns_for(data), fields, used_fields(fields, extra_fields)}
```

Comment:
```suggestion
    used_fields = Enum.uniq(used_fields(fields) ++ List.wrap(extra_fields)))
    {columns_for(data), fields, used_fields}
```

And then you can remove the uniq: true from the comprehension. :)

Reply:
And remove `used_fields/2`.